### ISSUE TITLE
stream$Readable should accept any type in object mode

### ISSUE THREAD

**Issue by mgtitimoli** [facebook/flow#3289]
Hi there,

I am using flow in a node project, and when I was extending a readable's stream I got the following error:

![image](https://cloud.githubusercontent.com/assets/4404683/22397935/f4a561f8-e55b-11e6-8bdb-ed5b18b2d553.png)

So I dug into the node's declaration file and I found this:

https://github.com/facebook/flow/blob/master/lib/node.js#L1246

Node's documentation is also wrong, since for `push` method it says that it only accepts:

`Buffer | null | String` (see [here](https://nodejs.org/api/stream.html#stream_readable_push_chunk_encoding))

But then when explains how **object mode** works (see [here](https://nodejs.org/api/stream.html#stream_object_mode)) it says what it really happens; that you can push any JS value if this mode is enabled

I was wondering what would be the best way to specify this behavior...

IMHO there are 2 alternatives:

1. Switch it to be a generic type depending on the options, and then have 2 different options types: one with object mode `true` and the other with `false`, as this will allow to specify what `push` accepts based on the options
2. Simply add `any` to the types it accepts
 
Any thoughts?

**Comment by philikon** [comment]
Dupe of #2010?

**Comment by mgtitimoli** [comment]
You are right @philikon, should I close the issue then (I don't know how you are dealing with duplicates)?

**Comment by philikon** [comment]
Yeah, I'd close this one in favor of the #2010.

**Comment by SamChou19815** [comment]
The libdef has been moved to flow-typed. If you want to make improvements, please make PR over there.
